forge/test/operators/utils/compat.py

```diff
@@ -22,6 +22,85 @@ class TestDevice:
     pass
 
 
+class TestTensorsUtils:
+
+    dev_data_format_to_dtype = {
+        # forge.DataFormat.Bfp2: torch.float16,
+        forge.DataFormat.Bfp2: torch.bool,  # mapped
+        forge.DataFormat.Bfp2_b: torch.bfloat16,
+        forge.DataFormat.Bfp4: torch.float16,
+        forge.DataFormat.Bfp4_b: torch.bfloat16,
+        forge.DataFormat.Bfp8: torch.float16,
+        forge.DataFormat.Bfp8_b: torch.bfloat16,
+
+        forge.DataFormat.Float16: torch.float16,  # mapped
+        forge.DataFormat.Float16_b: torch.bfloat16,  # mapped
+        forge.DataFormat.Float32: torch.float32,  # mapped
+
+        forge.DataFormat.Int8: torch.int8,  # mapped
+        forge.DataFormat.Lf8: torch.float16,
+        forge.DataFormat.RawUInt16: torch.uint8,
+        forge.DataFormat.RawUInt32: torch.uint8,
+        forge.DataFormat.RawUInt8: torch.uint8,
+        forge.DataFormat.UInt16: torch.uint8,
+        forge.DataFormat.Int32: torch.int32,  # mapped
+
+    }
+
+    @classmethod
+    def get_dtype_for_df(cls, dev_data_format: forge.DataFormat = None) -> torch.dtype:
+
+        dtype: torch.dtype
+
+        if dev_data_format is None:
+            dtype = None
+        else:
+            # dtype = torch.float32
+            if dev_data_format in cls.dev_data_format_to_dtype:
+                dtype = cls.dev_data_format_to_dtype[dev_data_format]
+            else:
+                raise ValueError(f"Unsupported dtype for dev data format: {dev_data_format}")
+
+            if dtype in (torch.uint8,):
+                dtype = torch.int8 
+
+        return dtype
+
+    @classmethod
+    def move_value_range(cls, inputs: List[torch.Tensor], min, max) -> List[torch.Tensor]:
+        inputs = [min + (max - min) * input for input in inputs]
+        return inputs
+
+    @classmethod
+    def get_random_sign(cls, generator: torch.Generator) -> int:
+        return 2 * torch.randint(0, 2, (1,), generator=generator).item() - 1
+
+    @classmethod
+    def get_random_torch_inputs(cls, dtype: torch.dtype, input_shapes: List[TensorShape], generator: torch.Generator) -> List[torch.Tensor]:
+
+        if dtype is None:
+            inputs = [torch.rand(input_shape, generator=generator) for input_shape in input_shapes]
+        elif dtype in (torch.float16, torch.bfloat16, torch.float32):
+            # TODO Testing big and small numbers should be configurable depending on operator
+            # Some operators support mix small and big numbers
+            # TODO Testing small numbers close to 0
+            inputs = [torch.rand(input_shape, dtype=dtype, generator=generator) for input_shape in input_shapes]
+            min, max = 100, 10000
```

Comment:
Why this range? It is probably most important to test inputs around 0. Are we doing this range (close to zero somewhere else?)

Reply:
We plan to test both small and big numbers. Depending on operator it should be configurable.

Reply:
Included initial implementation of value range limiting via ValueRanges.
It's now supported to specify different value range for input tensors depending on operator and/or dev data type.

E.x. for sqrt accepts only positive numbers, for some operators we can specify small (<1) or large (unlimited) value ranges